src/main/java/org/jabref/gui/mergeentries/newmergedialog/ThreeWayMergeView.java

```diff
@@ -0,0 +1,165 @@
+package org.jabref.gui.mergeentries.newmergedialog;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.RowConstraints;
+import javafx.scene.layout.VBox;
+import javafx.stage.Screen;
+
+import org.jabref.gui.Globals;
+import org.jabref.gui.mergeentries.newmergedialog.fieldsmerger.FieldMergerFactory;
+import org.jabref.gui.mergeentries.newmergedialog.toolbar.ThreeWayMergeToolbar;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+public class ThreeWayMergeView extends VBox {
+    public static final int GRID_COLUMN_MIN_WIDTH = 250;
+
+    public static final String LEFT_DEFAULT_HEADER = Localization.lang("Left Entry");
+    public static final String RIGHT_DEFAULT_HEADER = Localization.lang("Right Entry");
+
+    private final ColumnConstraints fieldNameColumnConstraints = new ColumnConstraints(150);
+    private final ColumnConstraints leftEntryColumnConstraints = new ColumnConstraints(GRID_COLUMN_MIN_WIDTH, 256, Double.MAX_VALUE);
+    private final ColumnConstraints rightEntryColumnConstraints = new ColumnConstraints(GRID_COLUMN_MIN_WIDTH, 256, Double.MAX_VALUE);
+    private final ColumnConstraints mergedEntryColumnConstraints = new ColumnConstraints(GRID_COLUMN_MIN_WIDTH, 256, Double.MAX_VALUE);
+    private final ThreeWayMergeToolbar toolbar;
+    private final ThreeWayMergeHeaderView headerView;
+    private final ScrollPane scrollPane;
+    private final GridPane mergeGridPane;
+
+    private final ThreeWayMergeViewModel viewModel;
+    private final List<FieldRowView> fieldRows = new ArrayList<>();
+
+    private final FieldMergerFactory fieldMergerFactory;
+
+    public ThreeWayMergeView(BibEntry leftEntry, BibEntry rightEntry, String leftHeader, String rightHeader) {
+        getStylesheets().add(ThreeWayMergeView.class.getResource("ThreeWayMergeView.css").toExternalForm());
+        viewModel = new ThreeWayMergeViewModel((BibEntry) leftEntry.clone(), (BibEntry) rightEntry.clone(), leftHeader, rightHeader);
+        // TODO: Inject 'preferenceService' into the constructor
```

Comment:
You can test if you can inject it as field
`@Inject private PreferencesService preferencesService`

Reply:
It throws a null pointer exception. I think the injection is handled by mvvmfx but `ThreeWayMergeView` doesn't use FXML or a view loader. I thought that because `@Inject` is used in other classes and all of them also use view loader.

Reply:
You need to inject it into DuplicateResolverDialog

Reply:
And also pass it from the  ChangeScanner ->  EntryChangeViewModel  -> ThreeWayMergeView via constructor